packages/core-p2p/src/server/versions/1/handlers.ts

```diff
@@ -180,7 +180,10 @@ export const postBlock = {
 
             return { success: true };
         } catch (error) {
-            logger.error(error);
+            //  Only write the error to the log if the error object is not empty
+            if(!!Object.keys(error).length) {
```

Comment:
What is the purpose of `!!`? Would the code behave in an identical way without it?

I would suggest to avoid treating a number like a boolean and do `Object.keys(error).length > 0` instead.

Missing space after `if`.